src/FSharp.Build/FSharpEmbedResourceText.fs

```diff
@@ -6,24 +6,32 @@ open System.IO
 open Microsoft.Build.Framework
 open Microsoft.Build.Utilities
 
-type FSharpEmbedResourceText() =
-    let mutable _buildEngine: IBuildEngine MaybeNull = null
-    let mutable _hostObject: ITaskHost MaybeNull = null
+/// A special exception that when thrown signifies that
```

Comment:
Is this really a better to do things here? To me, it feels a bit like a step toward using exceptions for flow control, not entirely yet though :) I might not have enough context so just asking to be sure.